src/edge.hpp

```diff
@@ -0,0 +1,17 @@
+#ifndef EDGE_HPP
+#define EDGE_HPP
+
+namespace graph {
+
+template<typename Node, typename... Types>
+class Edge : public Types... {
+public:
+    Edge(const Node& from, const Node& to, Types... args) : from(from), to(to), Types(args)... {}
```

Comment:
Pass by forwarding references and use `std::forward`:

`Types&&... args` => `Types(std::forward<Ts>(args))`

Reply:
Why would we force users to use ```std::move```?
```Types&&... args => Types(std::forward<Ts>(args))```

Reply:
Fixed